angular-workspace/projects/ni/nimble-angular/src/directives/table-column/anchor/nimble-table-column-anchor-router-link-with-href.directive.ts

```diff
@@ -0,0 +1,45 @@
+import { LocationStrategy } from '@angular/common';
+import { Directive, ElementRef, HostListener, Inject } from '@angular/core';
+import { ActivatedRoute, Router, RouterLinkWithHref } from '@angular/router';
+import type { TableColumnAnchor } from '@ni/nimble-components/dist/esm/table-column/anchor';
+import type { TableColumnAnchorCellView } from '@ni/nimble-components/dist/esm/table-column/anchor/cell-view';
+import type { DelegatedEventEventDetails } from '../base/nimble-table-column-base.directive';
+
+/**
+ * Directive to handle nimble-table-column-anchor RouterLink support.
+ * Note: Clients need to use [nimbleRouterLink] instead of [routerLink], so that there
+ * won't also be an active RouterLink directive incorrectly handling navigation.
+ */
+@Directive({
+    selector: 'nimble-table-column-anchor[nimbleRouterLink]'
+})
+export class NimbleTableColumnAnchorRouterLinkWithHrefDirective extends RouterLinkWithHref {
+    @HostListener('delegated-event', ['$event.details.originalEvent'])
+    private onDelegatedEvent(delegatedEvent: Event): void {
+        if (delegatedEvent.type !== 'click') {
+            return;
+        }
+
+        const clickEvent = delegatedEvent as MouseEvent;
+        const anchor = (delegatedEvent.target as TableColumnAnchorCellView).anchor;
+
+        if (!anchor) {
+            return;
+        }
+
+        // Only handle plain left-clicks
+        if (clickEvent.button !== 0 || clickEvent.ctrlKey || clickEvent.shiftKey || clickEvent.altKey || clickEvent.metaKey) {
+            return;
+        }
+
+        // Only handle when the target is the default: the current browsing context
+        if (typeof anchor.target === 'string' && anchor.target !== '_self') {
+            return;
+        }
```

Comment:
This logic is also present in `onClick`, but I was replicating it here so that I could know whether to let the router handle the navigation or let the default anchor action run. Looking again, I think I can just look at the return value of `onClick` and conditionally call `preventDefault` then instead. We don't have to worry about calling `preventDefault` explicitly in the breadcrumb case, because in that case, the return value of the click handler will do that implicitly. (It won't in this case because mine is handling the `delegated-event` event.)